pypeit/core/findobj_skymask.py

```diff
@@ -124,7 +124,18 @@ def create_skymask(sobjs, thismask, slit_left, slit_righ, box_rad_pix=None, trim
     # DP: I think skymask should always be skymask_objflux & skymask_fwhm (i.e., not only when box_rad_pix is not None).
     # In the case of skymask_objflux | skymask_fwhm, if skymask_objflux cannot be computed, the entire slit
     # is used for sky calculation (i.e., skymask_fwhm will not have effect).
-    skymask = skymask_objflux & skymask_fwhm
+
+    # DP's change which I don't think we should adopt at this time.
+    #skymask = skymask_objflux & skymask_fwhm
+
+    # JFH restored old behavior after seeing spurious results for X-shooter. I think the issue here is that the fwhm
+    # computation from objs_in_slit is not necessarily that reliable and when large amounts of masking are performed
+    # on narrow slits/orders, we have problems. We should revisit this after object finding is refactored since
+    # maybe then the fwhm estimates will be more robust.
+    if box_rad_pix is None:
+        skymask = skymask_objflux | skymask_fwhm
```

Comment:
Hey Joe, I still think the old behaviour is not good. As I explained above, if `skymask_objflux` cannot be computed (and it's the case for the maskdef extract objects because `smash_peakflux` and `THRESHOLD` are not available for those objects) `skymask_objflux | skymask_fwhm` will basically be equal to `thismask`, i.e., no masking is made. We should probably add another condition: if `skymask_objflux != thismask and  box_rad_pix is None` we use  `skymask = skymask_objflux | skymask_fwhm` otherwise we use  `skymask_objflux & skymask_fwhm`

Reply:
Ok @jhennawi. I can work on getting smash_peakflux and threshold for the maskdef extract objects. But for now, can we use the stopgap solution I suggested (or any other that work)?  Changing the condition `if box_rad_pix is None:` to `if box_rad_pix is None and np.any(skymask_objflux != thismask):` should work.

Reply:
I'm not sure I follow. Can we chat.